internal/backend/snapshot_messages.go

```diff
@@ -9,7 +9,7 @@ import (
 
 // snapMsg is a single message inside a snapshot.
 type snapMsg struct {
-	ID    string
+	ID    MessageIDPair
```

Comment:
I have a feeling we don't actually need the ID pair here, i can't think of a reason why the snapshot should care about the remote ID... can you?

if we could remove the remote ID here, that would simplify things even more -- can ignore my previous comment about the race condition, and we could remove the need to update remote IDs in the snapshot.

Reply:
It's mostly there to deal with caching data from the database. This can be replaced with a db lookup once we get closer to the connector for the actual request.